centroid_decompostition.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+class Graph 
+{
+private:
+	int ver;
+	vector<int>* adj;
+	vector<int>* centroidTree;
+	int* sizes;
+	bool* deleted;
+public:
+	Graph(int num) : ver(num) {
+		adj = new vector<int>(ver + 1);
```

Comment:
Any particular reason you are using ver + 1? (Don't immediately change anything, I just want to know if that is part of the algorithm)

Reply:
Since indexation starts from 0, you need additional space for Nth node